src/components/character-count/character-count.js

```diff
@@ -0,0 +1,179 @@
+import '../../vendor/polyfills/Function/prototype/bind'
+import '../../vendor/polyfills/Event' // addEventListener and event.target normaliziation
+
+function CharacterCount ($module) {
+  this.$module = $module
+  this.$textarea = $module.querySelector('.js-character-count')
+}
+
+CharacterCount.prototype.defaults = {
+  characterCountAttribute: 'data-maxlength',
+  wordCountAttribute: 'data-maxwords'
+}
+
+// Initialize component
+CharacterCount.prototype.init = function () {
+  // Check for module
+  var $module = this.$module
+  var $textarea = this.$textarea
+  if (!$textarea) {
+    return
+  }
+
+  // Read options set using dataset ('data-' values)
+  this.options = this.getDataset($module)
+
+  // Determine the limit attribute (characters or words)
+  var countAttribute = (this.options.maxwords) ? this.defaults.wordCountAttribute : this.defaults.characterCountAttribute
+
+  // Set the element limit
+  if ($module.getAttribute) {
+    this.maxLength = $module.getAttribute(countAttribute)
+  } else {
+    return
+  }
+
+  // Generate and reference message
+  var boundCreateCountMessage = this.createCountMessage.bind(this)
+  this.countMessage = boundCreateCountMessage()
+
+  // If there's a maximum length defined and the count message was successfully applied
+  if (this.maxLength && this.countMessage) {
+    // Replace maxlength attribute with data-maxlength to avoid hard limit
+    $module.setAttribute('maxlength', '')
```

Comment:
if this is what I think it is, it's relevant to something I was concerned about - if users don't use the macro they could well add maxlength to the texarea html tag themselves. We could remove it in code as it looks like we are doing here, and/or talk about this aspect in the guidance ('do not add 'maxlength' because ...)

Reply:
Yes, just spoke about this with @joelanman. This is removing the hard limit if any set (needs a bit of refactor and I'll use `removeAttribute` instead of setting an empty string). How do you feel about this approach @nickcolley?